app/controllers/application_controller.rb

```diff
@@ -233,7 +233,19 @@ def check_for_select_portal_user_type
     if request.format && request.format.html? && current_visitor && current_visitor.require_portal_user_type && !current_visitor.has_portal_user_type?
       unless session_sensitive_path
         flash.keep
-        redirect_to portal_user_type_selector_path
+
+        #
+        # TODO If there were some way render the underlying content of
+        # the omniauth_origin url beneath the modal signup popup, that would
+        # be nice.
+        #
+        # Also if this could be passed in, rather than stored in the
+        # session, that would. 
+        #
+        # Otherwise, this is not actually used. 
+        #
+        redirect_to portal_user_type_selector_path(omniauth_origin: session["omniauth.origin"])
```

Comment:
The props are passed down to the portal-pages signup component, which performs the redirect in the case of SSO signup completion. 

```
      //
      // For omniauth final step, simply redirect to omniauth_origin
      //
      if((studentData || teacherData) && this.props.omniauth) {
        console.log("INFO omniauth final step, redirect.", this.props);
        var data = this.state.studentData ? this.state.studentData : this.state.teacherData;
        window.location.href = data.omniauth_origin;
        return null;
      }

```